demo/src/faber.ts

```diff
@@ -0,0 +1,168 @@
+    }
+    return await this.agent.connections.getById(this.connectionRecordAliceId)
+  }
+
+  private async receiveConnectionRequest(invitation_url: string) {
+    const http = 'http://localhost:9000?c_i='
+    let invitationJson = invitation_url.replace(http, '')
+
+    try {
+      invitationJson = JsonEncoder.fromBase64(invitationJson)
+    } catch (e) {
+      console.log(`${Color.green}\nIt looks like your invitation link is not correctly formatted?\n${Color.reset}`)
+      return
+    }
+
+    const invitationMessage = JsonTransformer.fromJSON(invitationJson, ConnectionInvitationMessage)
+    return await this.agent.connections.receiveInvitation(invitationMessage)
+  }
+
+  private async waitForConnection(connectionRecord: ConnectionRecord) {
+    connectionRecord = await this.agent.connections.returnWhenIsConnected(connectionRecord.id)
+    console.log(`${Color.green}${Output.connectionEstablished}${Color.reset}`)
+    return connectionRecord.id
+  }
+
+  public async acceptConnection(invitation_url: string) {
+    const connectionRecord = await this.receiveConnectionRequest(invitation_url)
+    if (connectionRecord === undefined) {
+      return
+    }
+    this.connectionRecordAliceId = await this.waitForConnection(connectionRecord)
+  }
+
+  private printSchema(name: string, version: string, attributes: string[]) {
+    console.log(`\n\nThe credential definition will look like this:\n`)
+    console.log(`${Color.purlpe}Name: ${Color.reset}${name}`)
+    console.log(`${Color.purlpe}Version: ${Color.reset}${version}`)
+    console.log(`${Color.purlpe}Attributes: ${Color.reset}${attributes[0]}, ${attributes[1]}, ${attributes[2]}\n`)
+  }
+
+  private async registerSchema() {
+    const schemaTemplate = {
+      name: 'Faber College' + uuid(),
+      version: '1.0.0',
+      attributes: ['name', 'degree', 'date'],
+    }
+    this.printSchema(schemaTemplate.name, schemaTemplate.version, schemaTemplate.attributes)
+    this.ui.updateBottomBar(`${Color.green}\nRegistering schema...\n`)
+    const schema = await this.agent.ledger.registerSchema(schemaTemplate)
+    this.ui.updateBottomBar('\nSchema registerd!\n')
+    return schema
+  }
+
+  private async registerCredentialDefiniton(schema: Schema) {
+    this.ui.updateBottomBar('\nRegistering credential defenition...\n')
+    this.credentialDefinition = await this.agent.ledger.registerCredentialDefinition({
+      schema,
+      tag: 'latest',
+      supportRevocation: false,
+    })
+    this.ui.updateBottomBar('\nCredential definition registerd!!\n')
+    return this.credentialDefinition
+  }
+
+  private getCredentialPreview() {
+    const credentialPreview = CredentialPreview.fromRecord({
+      name: 'Alice Smith',
+      degree: 'Computer Science',
+      date: '01/01/2022',
+    })
+    return credentialPreview
+  }
+
+  public async issueCredential() {
+    const schema = await this.registerSchema()
+    const credentialDefinition = await this.registerCredentialDefiniton(schema)
```

Comment:
creating a schema and credential definition only needs to happen once, I think it would be good to reflect that in the demo.